mpack-instance-manager/src/main/python/instance_manager/instance_manager.py

```diff
@@ -149,12 +165,70 @@ def get_run_dir(mpack=None, mpack_instance=None, subgroup_name=DEFAULT_SUBGROUP_
   Components are provided as map with key as 'component type' and value as 'list of individual component instances
   names' OR empty map for all component instances present
   """
-  return build_granular_json_with_filtering(mpack, mpack_instance, subgroup_name, module_name, components_map,
-                                            output_run_dir=True)
+  dirs, is_client = get_conf_log_run_dir_helper(mpack, mpack_instance, subgroup_name, module_name, components_map, output_conf_dir = False, output_log_dir = False, output_run_dir = True)
+  if not is_client:
+    return [dir for dir in dirs if
+            (mpack == None or mpack in dir) and (mpack_instance == None or mpack_instance in dir) and (
+            module_name == None or module_name in dir)]
+  return [dir for dir in dirs if
+            (mpack == None or mpack in dir) and (mpack_instance == None or mpack_instance in dir)]
+
+
+def walk_mpack_dict(mpack_dict, filter, dirs):
+  """
```

Comment:
In multiple mpack instances scenario, we need traverse the whole mpack dict and get all the paths that we want.